src/components/ListItem.tsx

```diff
@@ -1,8 +1,70 @@
-import * as api from '../api';
-import './ListItem.css';
+import "./ListItem.css";
+import { updateItem, ListItem } from "../api";
+import { useEffect, useState } from "react";
+import toast from "react-hot-toast";
+import { is24HoursLater } from "../utils";
 
-type Props = Pick<api.ListItem, 'name'>;
+interface Props {
+	item: ListItem;
+	listPath: string | null;
+}
+
+export function ListItemCheckBox({ item, listPath }: Props) {
+	const [checked, setChecked] = useState(false);
+
+	const purchaseDate = item.dateLastPurchased
+		? item.dateLastPurchased.toDate()
+		: item.dateLastPurchased;
+
+	useEffect(() => {
+		if (purchaseDate) {
+			setChecked(!is24HoursLater(purchaseDate));
+		} else {
+			setChecked(false);
+		}
+	}, [item.dateLastPurchased]);
+
+	const handleCheckChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+		const isChecked = e.target.checked;
+
+		if (!isChecked && purchaseDate && !is24HoursLater(purchaseDate)) {
+			toast.error(
+				`${item.name} has already been marked as purchased in the last 24 hours.`,
```

Comment:
I wonder if this is really an error state 🤔 are we saying we want to physically prevent someone from buying things twice? I suppose it could be, but in a real world scenario it would be nice to see an override here. Like, "are you sure?" (that's 100% out of scope here, just food for thought)

Reply:
We had talked about more of a prompt that says "are you sure" of some type that will allow it to possibly still buy it twice but it did seem out of scope for this PR and we didn't want it to keep updating the DB when that may not be the intent. So I did think that having an error get thrown would something that is maybe a temporary solution.

Reply:
That makes sense 👍 I think @zahrafalak 's suggestion here could be a better way of handling this 😄 If they can't do this action without seeing an error, it's probably best to just not let them do it at all https://github.com/the-collab-lab/tcl-77-smart-shopping-list/pull/31/files#r1747943276